.github/workflows/main.yml

```diff
@@ -0,0 +1,587 @@
+
+      - name: 'Restore libluajit from cache'
+        id: libluajit-cache
+        uses: actions/cache@v2.1.2
+        env:
+          CACHE_NAME: 'libluajit-${{ matrix.arch }}'
+        with:
+          path: ${{ github.workspace }}/macos_build_temp/LuaJIT-${{ env.LIBLUAJIT_VERSION }}
+          key: ${{ runner.os }}-deps-${{ env.CACHE_NAME }}-${{ env.LIBLUAJIT_VERSION }}
+
+      - name: 'Build dependency libluajit'
+        if: steps.libluajit-cache.outputs.cache-hit != 'true'
+        run: CI/macos/build_libluajit.sh -s -a ${{ matrix.arch }}
+
+      - name: 'Install dependency libluajit'
+        if: steps.libluajit-cache.outputs.cache-hit == 'true'
+        run: CI/macos/build_libluajit.sh -s -a ${{ matrix.arch }} --install
+
+      - name: 'Restore libfreetype from cache'
+        id: libfreetype-cache
+        uses: actions/cache@v2.1.2
+        env:
+          CACHE_NAME: 'libfreetype-${{ matrix.arch }}'
+        with:
+          path: ${{ github.workspace }}/macos_build_temp/freetype-${{ env.LIBFREETYPE_VERSION }}
+          key: ${{ runner.os }}-deps-${{ env.CACHE_NAME }}-${{ env.LIBFREETYPE_VERSION }}
+
+      - name: 'Build dependency libfreetype'
+        if: steps.libfreetype-cache.outputs.cache-hit != 'true'
+        run: CI/macos/build_libfreetype.sh -s -a ${{ matrix.arch }}
+
+      - name: 'Install dependency libfreetype'
+        if: steps.libfreetype-cache.outputs.cache-hit == 'true'
+        run: CI/macos/build_libfreetype.sh -s -a ${{ matrix.arch }} --install
+
+      - name: 'Restore librnnoise from cache'
+        id: librnnoise-cache
+        uses: actions/cache@v2.1.2
+        env:
+          CACHE_NAME: 'librnnoise-${{ matrix.arch }}'
+        with:
+          path: ${{ github.workspace }}/macos_build_temp/rnnoise-${{ env.LIBRNNOISE_VERSION }}
+          key: ${{ runner.os }}-deps-${{ env.CACHE_NAME }}-${{ env.LIBRNNOISE_VERSION }}
+
+      - name: 'Build dependency librnnoise'
+        if: steps.librnnoise-cache.outputs.cache-hit != 'true'
+        run: CI/macos/build_librnnoise.sh -s -a ${{ matrix.arch }}
+
+      - name: 'Install dependency librnnoise'
+        if: steps.librnnoise-cache.outputs.cache-hit == 'true'
+        run: CI/macos/build_librnnoise.sh -s -a ${{ matrix.arch }} --install
+
+      - name: 'Package dependencies'
+        if: success()
+        working-directory: ${{ github.workspace }}/macos/obs-dependencies-${{ matrix.arch }}
+        run: |
+          CURRENT_DATE=$(date +"%Y-%m-%d")
+          FILE_NAME="macos-deps-${CURRENT_DATE}-${{ matrix.arch }}.tar.xz"
+          echo "FILE_NAME=${FILE_NAME}" >> $GITHUB_ENV
+
+          find . \( -type f -or -type l \) \( -name "*.la" -or -name "*.a" \) | xargs rm
+          find ./lib -mindepth 1 -maxdepth 1 -type d | xargs rm -rf
+          rm -rf ./bin
+          rm -rf ./share
+          cp -R ${{ github.workspace }}/licenses .
+
+          XZ_OPT=-T0 tar -cJf "${FILE_NAME}" *
+
+          mv ${FILE_NAME} ..
```

Comment:
Would this make more sense as its own script instead of raw script in yaml?

Reply:
It is implemented in the packaging scripts but those contain much more convenience functionality that wasn't needed here.